### PR TITLE
Adding missing unordered_set functions and refactoring the unordered_…

### PR DESCRIPTION
**Pull request by GustavooPaiva**
There we go, the missing functions. Also i tried to use tips from you guys, @xtofl and @Dobiasd.
Sadly it is not uniform as @xtofl proposed. I'll try to make it work in the next week or so.

### CODE REVIEW

File: include/fplus/sets.hpp [hunk 1]
```diff
@@ -31,12 +31,9 @@ bool set_includes(const SetType& set1, const SetType& set2)
...
 bool unordered_set_includes(const UnorderedSetType& set1, const UnorderedSetType& set2)
 {
-    for(const auto& x : set2)
-    {
-        if (set1.find(x) == set1.end())
-            return false;
-    }
-    return true;
+    auto first_not_included = std::find_if(set2.begin(), set2.end(),
+                                           [&](const typename UnorderedSetType::value_type& x) -> bool { return set1.find(x) == set1.end();});
```
Comment (xtofl) [inline include/fplus/sets.hpp]: style: this line goes over the Github diff window.

Also, there's no need to the lambda argument type or return type (supposing we do C++14):

    [&](auto&& x) { return set1.find(x) == set1.end();};

Comment (Dobiasd) [inline include/fplus/sets.hpp]: Up to now FunctionalPlus works with C++11 and does not need a C++14 compiler. I would like to keep it that way.

**Comment by Dobiasd** [comment]
Thanks @GustavooPaiva. Could you try to keep the code lines in the library code at a maximum of 80 characters, please?
For unit test code and API search types this rule does not apply.

**Comment by GustavooPaiva** [comment]
There you go, @Dobiasd.

**Comment by Dobiasd** [comment]
:+1:

**Pull request merged**